ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/metrics/LatencyMetric.scala

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.benchtool.metrics
+import com.daml.ledger.api.benchtool.metrics.LatencyMetric.{LatencyNanos, MaxLatency, Value}
+
+import java.time.Duration
+
+case class LatencyMetric(totalNanos: LatencyNanos, numberObservations: Int, maxLatency: MaxLatency)
+    extends Metric[LatencyNanos] {
+  override type V = LatencyMetric.Value
+  override type Objective = MaxLatency
+
+  override def onNext(value: LatencyNanos): Metric[LatencyNanos] =
+    copy(
+      totalNanos = totalNanos + value,
+      numberObservations = numberObservations + 1,
+    )
+
+  override def periodicValue(periodDuration: Duration): (Metric[LatencyNanos], Value) =
+    this -> currentAverage
+
```

Comment:
Wouldn't an average value of a time window be more useful than a cumulative average? Not sure about the requirements though.

Reply:
I didn't bother with that, since we don't have requirements for periodic values. At the beginning we would only look at the final value. Let's expand on it once we learn more about this flow.

Reply:
Works for me 👍